File: src/main/java/io/github/osamabmaq/tafqeetj/converters/Tafqeet.java
package io.github.osamabmaq.tafqeetj.converters;

import java.math.BigDecimal;
import java.math.BigInteger;

public class Tafqeet {
    private final IntegerToWordsConverter integerConverter = IntegerToWordsConverter.getInstance();
    private final DecimalToWordsConverter decimalConverter = DecimalToWordsConverter.getInstance();

    private static Tafqeet instance;

    private Tafqeet() {
    }

    public static synchronized Tafqeet getInstance() {
        if (instance == null)
            instance = new Tafqeet();
        return instance;
    }

    public String doTafqeet(int number) {
        return integerConverter.convert(number).toString();
    }

    public String doTafqeet(byte number) {
        return integerConverter.convert(number).toString();
    }

    public String doTafqeet(short number) {
        return integerConverter.convert(number).toString();
    }

    public String doTafqeet(long number) {
        return integerConverter.convert(number).toString();
    }

    public String doTafqeet(BigInteger number) {
        return integerConverter.convert(number).toString();
    }

    public String doTafqeet(BigDecimal number) {
        return decimalConverter.convert(number).toString();
    }
}

File: src/main/kotlin/tech/moussa/tafqeetk/converters/Tafqeet.kt
package tech.moussa.tafqeetk.converters

import java.math.BigDecimal
import java.math.BigInteger

class Tafqeet private constructor(){
    private val integerConverter = IntegerToWordsConverter.getInstance()
    private val decimalConverter = DecimalToWordsConverter.getInstance()

    fun doTafqeet(number: Int): String {
        return this.integerConverter.convert(number).toString()
    }

    fun doTafqeet(number: Byte): String {
        return this.integerConverter.convert(number).toString()
    }

    fun doTafqeet(number: Short): String {
        return this.integerConverter.convert(number).toString()
    }

    fun doTafqeet(number: Long): String {
        return this.integerConverter.convert(number).toString()
    }

    fun doTafqeet(number: BigInteger): String {
        return this.integerConverter.convert(number).toString()
    }

    fun doTafqeet(number: BigDecimal): String {
        return this.decimalConverter.convert(number).toString()
    }

    fun doTafqeet(number: Double): String {
        return this.decimalConverter.convert(number).toString()
    }

    companion object {
        private var instance: Tafqeet? = null

        @Synchronized
        fun getInstance(): Tafqeet {
            if (Tafqeet.instance == null) {
                Tafqeet.instance = Tafqeet()
            }
            return Tafqeet.instance!!
        }
    }
}